### DISCUSSION TITLE
请问用Behaviors库把触发的带参数的路由事件绑定到viewModel上，我在顶级元素上绑定，参数源来自很多子控件该怎么办呢？

### DISCUSSION THREAD

**Discussion by WWMMFFLLRR** [BYJRK/DotNet-Discussions#232] [WPF 开发]
<Behaviors:Interaction.Triggers>
       <Behaviors:EventTrigger EventName="UpDateParamter">
           <Behaviors:InvokeCommandAction Command="{Binding 
               RelativeSource={RelativeSource FindAncestor, 
               AncestorType={x:Type Window}}, Path=DataContext.VerifyCommand}"
                                          CommandParameter="" />
       </Behaviors:EventTrigger>
   </Behaviors:Interaction.Triggers>

我是一个grid里面有很多相同的自定义控件parambox，parambox里面自定义的一个路由事件UpDateParamter，触发路由事件时，如果不用绑定，在grid是用parambox.UpDateParamter += handler 这种方式，view的后台代码是实现是可以触发的。但是想用Behaviors库在grid上监听各个parambox的事件，然后调用ViewModel的command，参数没有办法绑定。请问这种情况怎么实现呢

**Comment by BYJRK** [comment]
你自定义的这个事件是属于自定义控件 `ParamBox` 的，`Grid` 上并没有这个事件，事件无法冒泡传递给 `Grid` 的。你只能把行为添加给每一个 `ParamBox`。或者你可以考虑将 `ParamBox` 各自的 `DataContext` 保存在 VM 中的一个集合上，然后用 `ItemsSource` 的方式批量生成。这样就可以把行为写在 `DataTemplate` 里面了

**Comment by WWMMFFLLRR** [comment]
好的，明白了，多谢大佬！
